src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1524,7 +1525,7 @@
         ;; Wrap the ancestors-fn so that we don't send internal facts such as NegationResult
         ;; to user-provided productions.  Note that this work is memoized inside fact-type->roots.
         wrapped-ancestors-fn (fn [fact-type]
-                               (if (isa? fact-type :clara.rules.engine/system-type)
+                               (if (isa? fact-type ISystemFact)
```

Comment:
Why did you use isa? here instead of instance?  This work ends up being memoized so it isn't as big of a deal as the fact-type-fn but I don't see a reason not to just use instance? since you're moving the JVM class dispatch here anyway.